app/controllers/api/user/subscription_targets_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Api
+  module User
+    class SubscriptionTargetsController < ApiController
+      include PublicUserAuthenticatable
+
+      no_caching
+
+      before_action :authenticate!
+      before_action :find_subscription
+
+      def show
+        if @subscription.nil?
+          render json: { message: 'No token was provided' }, status: :unauthorized
+          return
+        end
+
+        render json: serialize(targets)
+      end
+
+      private
+
+      def targets
+        if filter_params[:active_commodities_type].present?
+          TimeMachine.now do
+            codes = Api::User::ActiveCommoditiesService.new(@subscription).call[filter_params[:active_commodities_type]]
```

Comment:
filter param needs to be cast to a symbol to return the hash value from `ActiveCommoditiesService` otherwise `nil` is returned.

```suggestion
            codes = Api::User::ActiveCommoditiesService.new(@subscription).call[filter_params[:active_commodities_type].to_sym]
```